skbio/io/format/embl.py

```diff
@@ -0,0 +1,1497 @@
+                   'RA': 'AUTHORS',
+                   'RP': 'REFERENCE',
+                   'RC': 'REFERENCE_COMMENT',
+                   'RX': 'CROSS_REFERENCE',
+                   'RG': 'GROUP',
+                   'RT': 'TITLE',
+                   'RL': 'JOURNAL',
+                   # Cross references
+                   'DR': 'DBSOURCE',
+                   'CC': 'COMMENT',
+                   # features
+                   'FH': 'FEATURES',
+                   'FT': 'FEATURES',
+                   'SQ': 'ORIGIN',
+                   }
+
+# the original genbank _yield_section divides entries in sections relying on
+# spaces (the same section has the same level of indentation). Uniprot entries
+# have a key for each line, so to divide record in sections I need to define a
+# corresponance for each key to section, then I will divide a record in
+# sections using these section name. Some keys are commented out since I don't
+# find them in example. What I have to do if I find them?
+KEYS_2_SECTIONS = {
+                   # identification
+                   'ID': 'LOCUS',
+                   'AC': 'ACCESSION',
+                   # PA means PARENT ACCESSION (?) and applies to
+                   # feature-level-products entries
+                   'PA': 'PARENT_ACCESSION',
+                   'PR': 'PROJECT_IDENTIFIER',
+                   'DT': 'DATE',
+                   'DE': 'DEFINITION',
+                   'GN': 'GENE_NAME',  # uniprot specific
+                   'KW': 'KEYWORDS',
+                   # Source (taxonomy and classification)
+                   'OS': 'SOURCE',
+                   'OC': 'SOURCE',
+                   'OG': 'SOURCE',
+                   # reference keys
+                   'RA': 'REFERENCE',
+                   'RP': 'REFERENCE',
+                   'RC': 'REFERENCE',
+                   'RX': 'REFERENCE',
+                   'RG': 'REFERENCE',
+                   'RT': 'REFERENCE',
+                   'RL': 'REFERENCE',
+                   'RN': 'SPACER',
+                   # Cross references
+                   'DR': 'DBSOURCE',
+                   'CC': 'COMMENT',
+                   'AH': 'ASSEMBLY',
+                   'AS': 'ASSEMBLY',
+                   'FH': 'FEATURES',
+                   'FT': 'FEATURES',
+                   # sequence
+                   'SQ': 'ORIGIN',
+                   '  ': 'ORIGIN',
+                   # 'CO': 'ORIGIN,'
+                   # spacer (discarded)
+                   'XX': 'SPACER'
+                  }
+
+
+# for convenience: I think such functions are more readadble whiel accessing
+# valuies in lambda functions
+def _get_embl_key(line):
+    """Return first part of a string as a embl key (ie 'AC M14399;' -> 'AC')"""
+
+    # embl keys have a fixed size of 2 chars
+    return line[:2]
+
+
+# get embl key from value
+# http://stackoverflow.com/questions/8023306/get-key-by-value-in-dictionary
+def _get_embl_key_by_value(value, mydict=KEYS_TRANSLATOR):
```

Comment:
Ok removed. But the problems persists:
```python
In [1]: d = {'foo': 'bar'}

In [2]: def f(x):
   ...:     x['foo'] = 'baz'
   ...:     

In [3]: f(d)

In [4]: d
Out[4]: {'foo': 'baz'}
```
This is because all objects (not strings) are passed by reference, so `d1=d` will point `d1` to the same memory location of `d`. This is why I modify the reference dictionary in `_parse_single_embl` but I create a copy of this dictionary in `_serialize_reference`, otherwise my data changes after the first serialization